staging/src/k8s.io/apimachinery/pkg/labels/selector.go

```diff
@@ -236,26 +236,29 @@ func (r *Requirement) hasValue(value string) bool {
 func (r *Requirement) Matches(ls Labels) bool {
 	switch r.operator {
 	case selection.In, selection.Equals, selection.DoubleEquals:
-		if !ls.Has(r.key) {
+		val, exist := ls.Lookup(r.key)
+		if !exist {
 			return false
 		}
-		return r.hasValue(ls.Get(r.key))
+		return r.hasValue(val)
 	case selection.NotIn, selection.NotEquals:
-		if !ls.Has(r.key) {
+		val, exist := ls.Lookup(r.key)
```

Comment:
You could reverse the existence test here, and return true if the key does not exist in the map.

```go
if val, ok := ls.Lookup(r.key); ok {
  return ...
}
return true
```